src/rgw/rgw_op.cc

```diff
@@ -3307,7 +3307,7 @@ void RGWCreateBucket::execute()
 					    placement_rule,
 					    &selected_placement_rule, nullptr);
     if (selected_placement_rule != s->bucket_info.placement_rule) {
-      op_ret = -EEXIST;
+      op_ret = -ERR_BUCKET_POLICY_CONFLICT;
```

Comment:
As per https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html#ErrorCodeList , seems like there is no such error "BucketPolicyConflict". As Matt mentioned, could you please provide relevant doc for reference. 

Also shouldn't we check here if its same user or different user trying to create the bucket to determine the proper error code to return?

Reply:
@soumyakoduri Thanks for reviewing.  sure, there is no such error "BucketPolicyConflict" in s3, but no one could describe the situation that the active s3 user want to create an  existing bucket with different Bucket Policy. @cbodley got me and gave me some suggestions, and I have modified and commit. Please review again, thanks!

Reply:
Changes look good to me. Are you planning to add a unit test case as well?